### PR TITLE
Changed documentation regarding extension properties api

### PR DESCRIPTION
**Pull request by Jolanrensen**
Fixes https://github.com/Kotlin/dataframe/issues/635
more explicit explanation of what happens in between cell calls in the extension properties api in notebooks

### CODE REVIEW

**Review by AndreiKingsley — CHANGES_REQUESTED** [review]
It seems to me that for very beginning users it may not be completely clear yet. I would add a DataFrame head (as table md) so the user would see its columns, then show 
1) that now df has properties with these names
2) show some simple example (select for example) with String API and say that now it can be rewritten like this with Column selection DSL.

File: docs/StardustDocs/topics/extensionPropertiesApi.md [hunk 1]
```diff
@@ -28,12 +39,9 @@ df.add("lastName") { name.split(",").last() }
...
-In notebooks, extension properties are generated for [`DataSchema`](schemas.md) that is extracted from [`DataFrame`](DataFrame.md) 
-instance after REPL line execution. 
-After that [`DataFrame`](DataFrame.md)  variable is typed with its own [`DataSchema`](schemas.md), so only valid extension properties corresponding to actual columns in DataFrame will be allowed by the compiler and suggested by completion.
-
 Extension properties can be generated in IntelliJ IDEA using the [Kotlin Dataframe Gradle plugin](schemasGradle.md#configuration).
 
 <warning>
-In notebooks generated properties won't appear and be updated until the cell has been executed. It often means that you have to introduce new variable frequently to sync extension properties with actual schema
+In notebooks generated properties won't appear and be updated until the cell has been executed.
+It often means that you have to introduce new variable frequently to sync extension properties with actual schema.
```
Comment (zaleslaw) [inline docs/StardustDocs/topics/extensionPropertiesApi.md:46]: NOTE: ?

**Review by zaleslaw — APPROVED** [review]

File: docs/StardustDocs/topics/extensionPropertiesApi.md (hunk 1 above)
Comment (Jolanrensen) [reply docs/StardustDocs/topics/extensionPropertiesApi.md:46]: No need, the warning tags already make it look like this:
image

**Review by AndreiKingsley — CHANGES_REQUESTED** [review]
The paragraph assumes that the user 1) knows the "titanic" columns 2) has learned the String/CA API well. I think this is not a good approach. Instead of adding some explanations of the working mechanism, we should let the user understand what extension properties bring (and show that they are generated from columns). Imagine you make a notebook that you show to a newbie in DataFrame - first you load a table and display it, look at it, look at its columns. Then you can show that your `df` has extensional properties (i.e. write the code `df.age` and show what it will output). Then you show an example with String/Accessors API, and then rewrite it with EP API, and say that's how great it is.

**Comment by Jolanrensen** [comment]
> The paragraph assumes that the user 1) knows the "titanic" columns 2) has learned the String/CA API well. I think this is not a good approach. Instead of adding some explanations of the working mechanism, we should let the user understand what extension properties bring (and show that they are generated from columns). Imagine you make a notebook that you show to a newbie in DataFrame - first you load a table and display it, look at it, look at its columns. Then you can show that your `df` has extensional properties (i.e. write the code `df.age` and show what it will output). Then you show an example with String/Accessors API, and then rewrite it with EP API, and say that's how great it is.

While that's a fantastic approach for, say, a demo, I'm not sure we can convey it like that on a static website. It might take some actual screenshots to show how generated accessors will look. Not a bad idea though, I'll think about it.

**Comment by Jolanrensen** [comment]
@AndreiKingsley so more like here? https://kotlinlang.org/docs/data-analysis-work-with-data-sources.html#display-data

**Comment by AndreiKingsley** [comment]
In this case there is no need to make a guide, a short chain explanation is enough: there was a dataframe with such columns -> such EPs were generated  -> they can be used instead of string/accessors and have a lot of advantages.

**Comment by Jolanrensen** [comment]
I made it simpler, now I'm just sending the users to https://kotlin.github.io/dataframe/schemas.html which explains everything with examples and all :)

**Review by AndreiKingsley — APPROVED** [review]

**Pull request merged**